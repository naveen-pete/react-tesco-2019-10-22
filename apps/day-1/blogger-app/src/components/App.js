import React from 'react';

import Header from './Header';
import Posts from './Posts';
import Footer from './Footer';

// ES6 / ES2015 - arrow function
const App = () => {
  return <div className="container">
    <Header />
    <Posts />
    <Footer />
  </div>;
};

export default App;
