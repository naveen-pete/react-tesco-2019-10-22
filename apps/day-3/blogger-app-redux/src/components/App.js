import React, { Component } from 'react';
import { Route, Switch } from 'react-router-dom';
import { connect } from 'react-redux';

import Header from './Header';
import Posts from './Posts';
import Footer from './Footer';
import Home from './Home';
import Nav from './Nav';
import PostForm from './PostForm';
import PostDetail from './PostDetail';
import { getCategories } from '../redux/actions';

class App extends Component {
  // state = {
  //   categories: []
  // }

  componentDidMount() {
    // getCategories()
    //   .then(categories => this.setState({ categories }))
    //   .catch(error => {
    //     console.log('Get categories failed.');
    //     console.log('Error:', error);
    //   });

    this.props.getCategories();
  }

  render() {
    // const { categories } = this.state;

    return <div className="container">
      <Header />
      <Nav />

      <Switch>
        {/* http://localhost:3000/ */}
        <Route exact path="/" component={Home} />

        {/* http://localhost:3000/posts */}
        <Route exact path="/posts" component={Posts} />

        {/* http://localhost:3000/posts/create */}
        <Route exact path="/posts/create" render={(props) => {
          return <PostForm {...props} />
        }} />

        {/* http://localhost:3000/posts/2 */}
        <Route exact path="/posts/:id" component={PostDetail} />

      </Switch>

      <Footer />
    </div>;
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    getCategories: () => dispatch(getCategories())
  }
};

export default connect(null, mapDispatchToProps)(App);
