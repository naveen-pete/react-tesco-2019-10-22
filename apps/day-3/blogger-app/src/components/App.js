import React from 'react';

import Header from './Header';
import Posts from './Posts';
import Footer from './Footer';
import Home from './Home';
import Nav from './Nav';

// ES6 / ES2015 - arrow function
const App = () => {
  return <div className="container">
    <Header />
    <Nav />
    <Home />
    <Posts />
    <Footer />
  </div>;
};

export default App;
