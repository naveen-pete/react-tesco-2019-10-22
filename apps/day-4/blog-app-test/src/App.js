import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Header from './components/common/header';
import PostsRoot from './components/posts/posts-root';
import PostDetail from './components/posts/post-detail';
import PostForm from './components/posts/post-form';

const App = (props) => {
  return (
    <div>
      <Header />
      <div className="container">
        <Switch>
          <Route path="/posts/new" component={PostForm} />
          <Route path="/posts/:id/edit" component={PostForm} />
          <Route path="/posts/:id" component={PostDetail} />
          <Route path="/" component={PostsRoot} />
        </Switch>
      </div>
    </div>
  );
};

export default App;
