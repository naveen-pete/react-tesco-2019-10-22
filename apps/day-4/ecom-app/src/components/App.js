import React from 'react';
import { Switch, Route } from 'react-router-dom';

import Header from './Header';
import Home from './Home';
import Products from './Products';
import ProductForm from './ProductForm';
import ProductDetail from './ProductDetail';
import Users from './Users';
import RegisterUser from './RegisterUser';
import Login from './Login';
import PrivateRoute from './PrivateRoute';

// Function based component
const App = () => {
  return <div className="container">
    <Header />
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/products" component={Products} />
      <PrivateRoute exact path="/products/new" component={ProductForm} />
      <Route exact path="/products/:id" component={ProductDetail} />
      <PrivateRoute path="/products/:id/edit" component={ProductForm} />
      <PrivateRoute admin="true" path="/users" component={Users} />
      <Route path="/register" component={RegisterUser} />
      <Route path="/login" component={Login} />
    </Switch>
  </div>;
};

export default App;
