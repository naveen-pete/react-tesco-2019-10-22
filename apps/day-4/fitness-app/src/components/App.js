import React, { Component } from 'react';
import Header from './Header';
import Exercises from './Exercises';
import Footer from './Footer';
import { categories, exercises } from '../store';

class App extends Component {
  state = {
    categories,
    exercises,
    exercise: {},
    category: ''
  }

  getExercisesByCategory = () => {
    const { exercises } = this.state;

    return Object.entries(exercises.reduce((acc, e) => {
      const { muscles } = e;
      acc[muscles] = acc[muscles] ? [...acc[muscles], e] : [e];
      return acc;
    }, {}));
  }

  handleExerciseSelect = (exercise) => {
    this.setState({
      exercise,
      editMode: false
    });
  }

  handleCategorySelect = (category) => {
    this.setState({
      category
    });
  }

  handleExerciseCreate = (exercise) => {
    this.setState(({ exercises }) => {
      return {
        exercises: [
          ...exercises,
          exercise
        ]
      };
    });
  }

  handleExerciseEdit = id => {
    this.setState(({ exercises }) => ({
      exercise: exercises.find(e => e.id === id),
      editMode: true
    }));
  }

  handleExerciseEditSubmit = exercise => {
    this.setState(({ exercises }) => ({
      exercises: exercises.map(e => e.id === exercise.id
        ? { ...exercise } : { ...e }),
      exercise,
      editMode: false
    }));
  }


  render() {
    const { categories, exercise, category, editMode } = this.state;
    const exercises = this.getExercisesByCategory();

    return (
      <div>
        <Header
          categories={categories}
          onExerciseCreate={this.handleExerciseCreate}
        />
        <Exercises
          categories={categories}
          exercises={exercises}
          exercise={exercise}
          category={category}
          editMode={editMode}
          onExerciseSelect={this.handleExerciseSelect}
          onExerciseEdit={this.handleExerciseEdit}
          onExerciseEditSubmit={this.handleExerciseEditSubmit}
        />
        <Footer
          categories={categories}
          onCategorySelect={this.handleCategorySelect}
          category={category}
        />
      </div>
    );
  }
}

export default App;
