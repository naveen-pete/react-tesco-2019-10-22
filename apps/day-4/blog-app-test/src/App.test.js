import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import {shallow} from 'enzyme'; 
import App from './App';

test('should render App correctly (using react-test-renderer)', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<App />);
  expect(renderer.getRenderOutput()).toMatchSnapshot();
});

test('should render App correctly (using enzyme)', () => {
  const wrapper = shallow(<App />);
  expect(wrapper).toMatchSnapshot();
});
