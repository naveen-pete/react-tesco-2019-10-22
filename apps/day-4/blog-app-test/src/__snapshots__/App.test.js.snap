// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App correctly (using enzyme) 1`] = `
<div>
  <Header />
  <div
    className="container"
  >
    <Switch>
      <Route
        component={[Function]}
        path="/posts/new"
      />
      <Route
        component={[Function]}
        path="/posts/:id/edit"
      />
      <Route
        component={[Function]}
        path="/posts/:id"
      />
      <Route
        component={[Function]}
        path="/"
      />
    </Switch>
  </div>
</div>
`;

exports[`should render App correctly (using react-test-renderer) 1`] = `
<div>
  <Header />
  <div
    className="container"
  >
    <Switch>
      <Route
        component={[Function]}
        path="/posts/new"
      />
      <Route
        component={[Function]}
        path="/posts/:id/edit"
      />
      <Route
        component={[Function]}
        path="/posts/:id"
      />
      <Route
        component={[Function]}
        path="/"
      />
    </Switch>
  </div>
</div>
`;
